### PR TITLE
Limit search by recipient to 50 results

### PR DESCRIPTION
**Pull request by quis**
Things could get ugly if you use a short search string on a service with lots of notifications…

### CODE REVIEW

File: app/dao/notifications_dao.py [hunk 1]
```diff
@@ -634,7 +634,11 @@ def dao_get_notifications_by_recipient_or_reference(service_id, search_term, not
     if notification_type:
         filters.append(Notification.notification_type == notification_type)
 
-    results = db.session.query(Notification).filter(*filters).order_by(desc(Notification.created_at)).all()
+    results = db.session.query(Notification)\
+        .filter(*filters)\
+        .order_by(desc(Notification.created_at))\
+        .limit(current_app.config['PAGE_SIZE'])\
```
Comment (idavidmcdonald) [inline app/dao/notifications_dao.py:640]: Optional: To pass the limit as an argument to the `dao` function and so it's more the view function that decides how many results should be retrieved which is more business logic (unless this a pattern that is already well established or other reasons)?

**Comment by quis** [comment]
I think having a message saying ‘Only showing the first 50 results‘ would be fine – chances are if you have more than 50 you need to make your search more specific. So maybe this endpoint should limit at 51 results so we can know there are more than 50? Or is that too dirty a way of doing it?

**Comment by idavidmcdonald** [comment]
Yeah, and only showing the first 50 is probably a quick solution for this case.

I think limiting to 51 isn't a bad shout, the question is just do you
-  send the 51 to the admin app and show the warning
-  send 50 and an extra variable saying more than 50 results.
- just sent 50 regardless of if there are more and then the admin shows the warning anyway 

I think I lean towards one of the later two as it feels weird to send 51 results across but I'd be happy with any really.

File: app/service/rest.py [hunk 2]
```diff
@@ -475,10 +475,18 @@ def search_for_notification_by_to_field(service_id, search_term, statuses, notif
...
         statuses=statuses,
-        notification_type=notification_type
+        notification_type=notification_type,
+        page=1,
+        page_size=current_app.config['PAGE_SIZE'],
     )
     return jsonify(
-        notifications=notification_with_template_schema.dump(results, many=True).data
+        notifications=notification_with_template_schema.dump(results.items, many=True).data,
+        links=pagination_links(
```
Comment (idavidmcdonald) [inline app/service/rest.py:484]: Optional: We have 4 other places we are returning paginated results. Three of the four seem to also return keys for total and page_size. Would be good to make it consistent I think?

Do we also want a unit test to check one of the responses at least contains the links key? I'm not sure if I'm too interested in testing the values of it as don't want to retest pagination_links itself but just as an indicator that we are returning paginated results?

**Pull request merged**